test/liblangutil/CharStream.cpp

```diff
@@ -28,6 +28,25 @@
 
 #include <boost/test/unit_test.hpp>
 
+using namespace std;
+
+namespace std
+{
+template<typename T>
+std::ostream& operator<<(std::ostream& _ostream, std::optional<T> const& _value)
+{
+	if (_value)
+		_ostream << *_value;
+	else
+		_ostream << "[nullopt]";
+	return _ostream;
+}
+inline std::ostream& operator<<(std::ostream& _ostream, std::nullopt_t const&)
+{
+	return _ostream << "[nullopt]";
+}
+}
```

Comment:
Given that we do not indent namespaces, I think it's a nice convention to comment where they end:

```suggestion

} //namespace std
```

Also, some whitespace after opening and before ending brace would be nice for readability.